File: api/src/modules/workspaces/entities/workspace-automation-config.entity.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  UpdateDateColumn,
  ManyToOne,
  JoinColumn,
} from 'typeorm';
import { Workspaces } from './workspaces.entity';

@Entity({ name: 'workspace_automation_configs' })
export class WorkspaceAutomationConfig {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ name: 'workspace_id', type: 'uuid' })
  workspaceId: string;

  @Column({ name: 'is_active', type: 'boolean', default: false })
  isActive: boolean;

  @Column({ type: 'text', default: 'America/New_York' })
  timezone: string;

  @Column({ name: 'generate_at', type: 'text', default: '19:00' })
  generateAt: string;

  @Column({ name: 'posts_per_cycle', type: 'int', default: 3 })
  postsPerCycle: number;

  @Column({ name: 'plan_ahead_days', type: 'int', default: 1 })
  planAheadDays: number;

  @Column({ name: 'publishing_days', type: 'jsonb', default: '[]' })
  publishingDays: string[];

  @Column({ name: 'posting_times', type: 'jsonb', default: '[]' })
  postingTimes: string[];

  @Column({ type: 'jsonb', default: '[]' })
  platforms: string[];

  @CreateDateColumn({ name: 'created_at', type: 'timestamptz' })
  created_at: Date;

  @UpdateDateColumn({ name: 'updated_at', type: 'timestamptz' })
  updated_at: Date;

  @ManyToOne(() => Workspaces, { nullable: false, onDelete: 'CASCADE' })
  @JoinColumn({ name: 'workspace_id' })
  workspace: Workspaces;
}

File: api-rust/src/modules/workspaces/workspace_automation_config_entity.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "workspace_automation_configs")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: Uuid,
    pub workspace_id: Uuid,
    pub is_active: bool,
    #[sea_orm(column_type = "Text")]
    pub timezone: String,
    #[sea_orm(column_type = "Text")]
    pub generate_at: String,
    pub posts_per_cycle: i32,
    pub plan_ahead_days: i32,
    pub publishing_days: Json,
    pub posting_times: Json,
    pub platforms: Json,
    pub created_at: DateTimeWithTimeZone,
    pub updated_at: DateTimeWithTimeZone,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(
        belongs_to = "super::entity::Entity",
        from = "Column::WorkspaceId",
        to = "super::entity::Column::Id"
    )]
    Workspace,
}

impl Related<super::entity::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::Workspace.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}
